docs/security/outputs.md

```diff
@@ -6,29 +6,57 @@ Untrusted data placed in HTML must be [encoded](index.md#encoding) for HTML outp
 
 ### Between tags {:#db-between-tags}
 
+Whether data needs to be output encoded depends primarily on the data source:
+
 #### Database data between tags {:#db-between-tags}
 
-Data which comes out of MySQL has already been [partially encoded for HTML output](inputs.md#input-encoding). This means that when you place this data between HTML tags, you don't need to perform any output encoding. For example:
+Data which comes *directly* out of MySQL has already been [partially encoded for HTML output](inputs.md#input-encoding). This means that when you place this data between HTML tags, you don't need to perform any output encoding. For example:
```

Comment:
@pfigel if i run `INSERT INTO civicrm_contact ('display_name') VALUES ('<p onmouseover=alert(\'hi\')>I am evil</p>');`  Then it isn't encoded in the database right? So its not just about the data source but how the data got in there in the first place right?

Reply:
This is an interesting point. Civi has this contract where [inputs are encoded](https://docs.civicrm.org/dev/en/latest/security/inputs/#input-encoding) prior to storage, so arguably, the vulnerability here would be in the code bypassing the input coder (even if that's a silly approach). It is something that's worth pointing out - I'll try to work in some language to make clear that only data that has previously gone through the input coder is "secure" in this context.